src/ProxyManager/ProxyGenerator/AccessInterceptorScopeLocalizer/MethodGenerator/BindProxyProperty.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ProxyManager\ProxyGenerator\AccessInterceptorScopeLocalizer\MethodGenerator;
+
+use Laminas\Code\Generator\ParameterGenerator;
+use ProxyManager\Exception\UnsupportedProxiedClassException;
+use ProxyManager\Generator\MethodGenerator;
+use ReflectionClass;
+
+use function strtr;
+
+class BindProxyProperty extends MethodGenerator
+{
+    public function __construct(ReflectionClass $originalClass)
+    {
+        parent::__construct(
+            'bindProxyProperty',
+            [
+                new ParameterGenerator('localizedObject', $originalClass->getName()),
+                new ParameterGenerator('class', ReflectionClass::class),
+                new ParameterGenerator('propertyName', 'string'),
+            ],
+            self::FLAG_PRIVATE
+        );
+
+        $bodyTemplate = <<<'CODE'
+        $originalClass = $class;
+        while (! $class->hasProperty($propertyName)) {
```

Comment:
This operation should be performed during compilation, rather than during runtime: it's quite expensive to traverse ancestors.

Reply:
@Ocramius unfortunately I don't understand how to perform this during compilation: different properties would have different declaring classes. So how this can be done during compilation?

Reply:
Done - I'm generating a `$classesMap` variable at build time, which has instances of `ReflectionClass` for private properties, and then use it when localizing private props.